evaluation/evaluator.go

```diff
@@ -26,6 +26,9 @@ const (
 	equalSensitiveOperator = "equal_sensitive"
 )
 
+// ProcessEvaluation is HOF for after evaluation processing
+type ProcessEvaluation func(fc *rest.FeatureConfig, target *Target, variation *rest.Variation)
```

Comment:
The usage of this ProcessEvaluation feels a little strange to me, when reading the code.
It looks like the only place we use it is from the Clients BoolVariation,StringVariation functions (where we pass 

c.analyticsService.PushToQueue to the relevant method of the Evaluator.

Can we separate evaluation from posting metrics, and call this from the Client?   

Another option might be for the Evaluator to manage a channel which the Client could listen on.  Any time an evaluation is successfully complete the evaluator can write to the channel.
The Client can have a routine somewhere reading from the channel and processing the metrics.